### PR TITLE
Upgrade simple history package

### PR DESCRIPTION
**Pull request by UsamaSadiq**
### Description
- PR created under the issue https://github.com/edx/edx-arch-experiments/issues/462
- Added common_constraints.txt to override global simple-history package constraint
- Bumped the django-simple-history package version to the latest
- Added new migrations caused by the new version of django-simple-history package.
- Added new setting to avoid db_indexing in the new migrations.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -102,8 +102,18 @@ html_coverage: ## generate and view HTML coverage report
...
+COMMON_CONSTRAINTS_TXT=requirements/common_constraints.txt
+.PHONY: $(COMMON_CONSTRAINTS_TXT)
+$(COMMON_CONSTRAINTS_TXT):
+	wget -O "$(@)" https://raw.githubusercontent.com/edx/edx-lint/master/edx_lint/files/common_constraints.txt || touch "$(@)"
+	echo "$(COMMON_CONSTRAINTS_TEMP_COMMENT)" | cat - $(@) > temp && mv temp $(@)
+
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
-upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+upgrade: $(COMMON_CONSTRAINTS_TXT) ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
+	sed 's/Django<2.3//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (iamsobanjaved) [inline Makefile:113]: Do we need this?

Comment (UsamaSadiq) [reply Makefile:113]: The Django sed command can be removed. Let me remove it.

Comment (UsamaSadiq) [inline Makefile:113]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -102,8 +102,18 @@ html_coverage: ## generate and view HTML coverage report
...
+	mv requirements/common_constraints.tmp requirements/common_constraints.txt
```
Comment (UsamaSadiq) [inline Makefile:114]:
```suggestion
```

**Comment by macdiesel** [comment]
@UsamaSadiq Is this a blocker for Django 4.2?

**Comment by UsamaSadiq** [comment]
> @UsamaSadiq Is this a blocker for Django 4.2?

Yes, this would need to be merged before upgrading to Django 4.2

**Review by macdiesel — APPROVED** [review]

**Pull request merged**